src/fixes/unordered_key.rs

```diff
@@ -279,6 +327,236 @@ mod tests {
         assert_lines(&lines, vec!["A=B", "B=C", "D=E"]);
     }
 
+    #[test]
+    fn key_order_substitution_variable_test() {
+        let mut lines = get_lines(vec![
+            "KEY=VALUE",
+            "ABC=XYZ",
+            "FOO=$KEY",
+            "BOO=$FOO",
+            "XYZ=ABC",
+            "BAR=FOO",
+        ]);
+
+        let mut warnings = get_warnings(
+            &lines,
+            vec![
+                (1, "The ABC key should go before KEY key"),
+                (5, "The BAR key should go before BOO key"),
+            ],
+        );
+
+        assert_eq!(Some(2), run_fixer(&mut warnings, &mut lines));
+
+        assert_lines(
+            &lines,
+            vec![
+                "ABC=XYZ",
+                "KEY=VALUE",
+                "FOO=$KEY # Unordered because FOO uses KEY",
+                "BAR=FOO",
+                "BOO=$FOO",
+                "XYZ=ABC",
+            ],
+        );
+    }
+
+    #[test]
+    fn key_order_substitution_variable_multiple_groups_test() {
+        let mut lines = get_lines(vec![
+            "KEY=VALUE",
+            "ABC=XYZ",
+            "FOO=$KEY",
+            "BOO=$FOO",
+            "XYZ=ABC",
+            "BAR=FOO",
+            "",
+            "M=1",
+            "N=2",
+            "A=$M",
+            "B=3",
+        ]);
+
+        let mut warnings = get_warnings(
+            &lines,
+            vec![
+                (1, "The ABC key should go before KEY key"),
+                (5, "The BAR key should go before BOO key"),
+                (9, "The A key should go before N key"),
```

Comment:
If I run `dotenv-linter` on this file, the output is as follows:
```
UnorderedKey: The ABC key should go before the KEY key
UnorderedKey: The BAR key should go before the BOO key

Found 2 problems
```

Reply:
Yeah, really good point 👍 
I'll put my two cents if you don't mind 🙂 

`UnorderedKeyChecker` already knows what these orders are correct (and the `Fixer` doesn't fix this in the current implementation, according to below asserts, i.e it works as expected).

I think this test should be left as is, except for this warning: 

```suggestion
```

Reply:
This is indeed a mistake on my end.